### PR TITLE
DSC19-112: Case Report map additional fields to DHIS

### PR DESCRIPTION
**Pull request by de-laz**

### CODE REVIEW

File: docker/importer/volume/surveillance-mediator/case-report/endpoint-resource-resolver-dhis.json [hunk 1]
```diff
@@ -82,7 +104,29 @@
     "patcourse_asymp": "{ 'dataElement': constants.dataElements.patcourse_asymp, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[3].value.valueCodeableConcept.coding[0].code) }",
     "Comcond_any": "{ 'dataElement': constants.dataElements.Comcond_any, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[5].value.valueCodeableConcept.coding[0].code) }",
     "expo_travel": "{ 'dataElement': constants.dataElements.expo_travel, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[6].value.valueCodeableConcept.coding[0].code) }",
-    "patcourse_dateonset": "{ 'dataElement': constants.dataElements.patcourse_dateonset, 'value': requestBody.data.observations[4].value.valueDate }"
+    "patcourse_dateonset": "{ 'dataElement': constants.dataElements.patcourse_dateonset, 'value': requestBody.data.observations[4].value.valueDate }",
```
Comment (bradsawadye) [inline docker/importer/volume/surveillance-mediator/case-report/endpoint-resource-resolver-dhis.json]:
```suggestion
    "patcourse_dateonset": "{ 'dataElement': constants.dataElements.patcourse_dateonset, 'value': requestBody.data.observations[4].value.valueDateTime }",
```

File: docker/importer/volume/surveillance-mediator/case-report/endpoint-resource-resolver-dhis.json [hunk 2]
```diff
@@ -82,7 +104,29 @@
...
+    "expo_contact_case": "{'dataElement': constants.dataElements.expo_contact_case, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[17].value.valueCodeableConcept.coding[0].code) }",
+    "expo_ID1": "{'dataElement': constants.dataElements.expo_ID1, 'value': request.data.observations[17].value.component[1].valueString}",
+    "expo_travel_country1": "{'dataElement': constants.dataElements.expo_travel_country1, 'value': request.data.observations[15].value.valueCodeableConcept.coding[0].code }",
+    "patcourse_icu": "{'dataElement': constants.dataElements.patcourse_icu, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[10].value.valueCodeableConcept.coding[0].code) }",
+    "patcourse_ecmo": "{'dataElement': constants.dataElements.patcourse_ecmo, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[12].value.valueCodeableConcept.coding[0].code) }",
+    "comcond_preg": "{'dataElement': constants.dataElements.comcond_preg, 'value': 'PREGNANCY' in request.data.conditions ? 'PREGNANCY' : '' }",
```
Comment (bradsawadye) [inline docker/importer/volume/surveillance-mediator/case-report/endpoint-resource-resolver-dhis.json]: the logic here will not work. `request.data.conditions` is an array containing objects not strings

Comment (de-laz) [reply docker/importer/volume/surveillance-mediator/case-report/endpoint-resource-resolver-dhis.json]: @bradsawadye What is the structure of the objects?

**Review by MattyJ007 — APPROVED** [review]
LGTM! 👍🏼 
Nice work

**Pull request merged**